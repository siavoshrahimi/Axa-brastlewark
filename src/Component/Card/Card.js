import React from 'react';
import {Link} from 'react-router-dom';

//css
import './Card.scss';

import CardDetail from "./card-detail/CardDetail";
import CardFriends from "./card-friends/CardFriends";
import CardProfession from "./card-profession/cardProfesion";

const card = ({allData,profileInfo}) =>{
  const {thumbnail,age,professions,friends,name} = profileInfo;
  const profileFriend = [];

  allData.map(profile => {
    friends.find(friend =>{
      if( friend === profile.name){
        profileFriend.push(profile);
        return profileFriend
      }
      return false
    });
    return true
  });
  return(

      <div className="card">
        <Link to={{pathname:`/detail/${name}`,state:{profileInfo}}}>
          <div className='img-wrapper'>
            <img className='img' src={thumbnail} alt={name}/>
          </div>
          <CardDetail name={name} age={age} />
        </Link>
          <div className='professions'>
            <span><strong>Professions:</strong></span>
            {professions.map(profession => <CardProfession key={profession}>{profession}</CardProfession>)}
          </div>
          <div className='friends'>
            <span>friends:</span>
            <div className='friends-img-wrapper'>
              {profileFriend[0] ?
                profileFriend.map(friend =><CardFriends friend={friend} key={friend.id}/>): <span>Has no friends</span>}
            </div>
          </div>
      </div>

  )
  
}

export default card;