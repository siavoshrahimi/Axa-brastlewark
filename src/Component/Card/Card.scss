.card{
  min-height: 300px;
  width: 30%;
  border: 1px solid #eee;
  box-shadow: 0 2px 3px #ccc;
  margin: 10px 0;
  border-radius: 3px;
  padding: 10px;
  transition: transform 0.2s linear;
  height:375px;
  &:hover{
    transform: translateY(-5px);
  }
  a{
    text-decoration: none;
    display: block;
    width: 100%;
    height: 216px;
    color:#000;
    position: relative;
    .img-wrapper{
      height: 100%;

      .img{
        height: 100%;
        width: 100%;
        border-radius: 3px;

      }
    }
    .card-detail{
      position: absolute;
      bottom:0;
      padding: 0 10px;
      left: 0;
      display: flex;
      justify-content: space-between;
      margin: 5px;
      font-size: 12px;
      width: 90%;
      span{
        padding: 5px 7px;
        background: #a3dfee;
        border-radius: 20px;
      }
    }
  }
  .professions{
    display: flex;
    flex-wrap: wrap;
    margin-top: 10px;
    .profession{
      padding: 2px 7px;
      background: #eee;
      border-radius: 10px;
      margin: 3px 5px;
      font-size: 13px;

    }
  }
  .friends{
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    height: 100px;
    width: 100%;
    margin-top: -5px;
    .friends-img-wrapper{
      a{
        text-decoration: none;
        display: inline;
      }
      .friends-img{
        border-radius: 50% 2px;
        width: 50px;
        height: 50px;
        margin-left: 10px;
        border: 1px solid #eee;
      }
    }
  }
}
