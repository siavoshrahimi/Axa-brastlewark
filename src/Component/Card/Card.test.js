import React from 'react';
import {shallow} from 'enzyme';

import Card from './Card';


it('expect to render card component' ,() =>{
    const mockAllData = [];
    const mockProfileInfo = {
        thumbnail:"jeff",
        age:23,
        professions:["metalWorker"],
        friends:[ 'david','jose'],
        name: 'maria'
    }
    expect(shallow(<Card profileInfo={mockProfileInfo} allData={mockAllData}/>)).toMatchSnapshot();
})