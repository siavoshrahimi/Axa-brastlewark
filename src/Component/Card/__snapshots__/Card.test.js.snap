// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render card component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <card
    allData={Array []}
    profileInfo={
      Object {
        "age": 23,
        "friends": Array [
          "david",
          "jose",
        ],
        "name": "maria",
        "professions": Array [
          "metalWorker",
        ],
        "thumbnail": "jeff",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Link
          to={
            Object {
              "pathname": "/detail/maria",
              "state": Object {
                "profileInfo": Object {
                  "age": 23,
                  "friends": Array [
                    "david",
                    "jose",
                  ],
                  "name": "maria",
                  "professions": Array [
                    "metalWorker",
                  ],
                  "thumbnail": "jeff",
                },
              },
            }
          }
        >
          <div
            className="img-wrapper"
          >
            <img
              alt="maria"
              className="img"
              src="jeff"
            />
          </div>
          <CardDetail
            age={23}
            name="maria"
          />
        </Link>,
        <div
          className="professions"
        >
          <span>
            <strong>
              Professions:
            </strong>
          </span>
          <CardProfession>
            metalWorker
          </CardProfession>
        </div>,
        <div
          className="friends"
        >
          <span>
            friends:
          </span>
          <div
            className="friends-img-wrapper"
          >
            <span>
              Has no friends
            </span>
          </div>
        </div>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              className="img-wrapper"
            >
              <img
                alt="maria"
                className="img"
                src="jeff"
              />
            </div>,
            <CardDetail
              age={23}
              name="maria"
            />,
          ],
          "to": Object {
            "pathname": "/detail/maria",
            "state": Object {
              "profileInfo": Object {
                "age": 23,
                "friends": Array [
                  "david",
                  "jose",
                ],
                "name": "maria",
                "professions": Array [
                  "metalWorker",
                ],
                "thumbnail": "jeff",
              },
            },
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="maria"
                className="img"
                src="jeff"
              />,
              "className": "img-wrapper",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "maria",
                "className": "img",
                "src": "jeff",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "age": 23,
              "name": "maria",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              <strong>
                Professions:
              </strong>
            </span>,
            Array [
              <CardProfession>
                metalWorker
              </CardProfession>,
            ],
          ],
          "className": "professions",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <strong>
                Professions:
              </strong>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Professions:",
              },
              "ref": null,
              "rendered": "Professions:",
              "type": "strong",
            },
            "type": "span",
          },
          Object {
            "instance": null,
            "key": "metalWorker",
            "nodeType": "function",
            "props": Object {
              "children": "metalWorker",
            },
            "ref": null,
            "rendered": "metalWorker",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span>
              friends:
            </span>,
            <div
              className="friends-img-wrapper"
            >
              <span>
                Has no friends
              </span>
            </div>,
          ],
          "className": "friends",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "friends:",
            },
            "ref": null,
            "rendered": "friends:",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <span>
                Has no friends
              </span>,
              "className": "friends-img-wrapper",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Has no friends",
              },
              "ref": null,
              "rendered": "Has no friends",
              "type": "span",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Link
            to={
              Object {
                "pathname": "/detail/maria",
                "state": Object {
                  "profileInfo": Object {
                    "age": 23,
                    "friends": Array [
                      "david",
                      "jose",
                    ],
                    "name": "maria",
                    "professions": Array [
                      "metalWorker",
                    ],
                    "thumbnail": "jeff",
                  },
                },
              }
            }
          >
            <div
              className="img-wrapper"
            >
              <img
                alt="maria"
                className="img"
                src="jeff"
              />
            </div>
            <CardDetail
              age={23}
              name="maria"
            />
          </Link>,
          <div
            className="professions"
          >
            <span>
              <strong>
                Professions:
              </strong>
            </span>
            <CardProfession>
              metalWorker
            </CardProfession>
          </div>,
          <div
            className="friends"
          >
            <span>
              friends:
            </span>
            <div
              className="friends-img-wrapper"
            >
              <span>
                Has no friends
              </span>
            </div>
          </div>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div
                className="img-wrapper"
              >
                <img
                  alt="maria"
                  className="img"
                  src="jeff"
                />
              </div>,
              <CardDetail
                age={23}
                name="maria"
              />,
            ],
            "to": Object {
              "pathname": "/detail/maria",
              "state": Object {
                "profileInfo": Object {
                  "age": 23,
                  "friends": Array [
                    "david",
                    "jose",
                  ],
                  "name": "maria",
                  "professions": Array [
                    "metalWorker",
                  ],
                  "thumbnail": "jeff",
                },
              },
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="maria"
                  className="img"
                  src="jeff"
                />,
                "className": "img-wrapper",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "maria",
                  "className": "img",
                  "src": "jeff",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "age": 23,
                "name": "maria",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                <strong>
                  Professions:
                </strong>
              </span>,
              Array [
                <CardProfession>
                  metalWorker
                </CardProfession>,
              ],
            ],
            "className": "professions",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <strong>
                  Professions:
                </strong>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Professions:",
                },
                "ref": null,
                "rendered": "Professions:",
                "type": "strong",
              },
              "type": "span",
            },
            Object {
              "instance": null,
              "key": "metalWorker",
              "nodeType": "function",
              "props": Object {
                "children": "metalWorker",
              },
              "ref": null,
              "rendered": "metalWorker",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                friends:
              </span>,
              <div
                className="friends-img-wrapper"
              >
                <span>
                  Has no friends
                </span>
              </div>,
            ],
            "className": "friends",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "friends:",
              },
              "ref": null,
              "rendered": "friends:",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <span>
                  Has no friends
                </span>,
                "className": "friends-img-wrapper",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Has no friends",
                },
                "ref": null,
                "rendered": "Has no friends",
                "type": "span",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
